java/ql/lib/semmle/code/java/frameworks/apache/Collections.qll

```diff
@@ -182,6 +204,695 @@ private class ApacheKeyValueModel extends SummaryModelCsv {
+          ".bag;TreeBag;true;TreeBag;(Collection);;Element of Argument[0];Element of Argument[-1];value",
+          ".bag;UnmodifiableBag;true;unmodifiableBag;;;Element of Argument[0];Element of ReturnValue;value",
+          ".bag;UnmodifiableSortedBag;true;unmodifiableSortedBag;;;Element of Argument[0];Element of ReturnValue;value"
+        ]
+  }
+}
+
+/**
+ * Value-propagating models for classes in the package `org.apache.commons.collections4.bidimap`.
+ */
+private class ApacheBidiMapModel extends SummaryModelCsv {
+  override predicate row(string row) {
+    row =
+      ["org.apache.commons.collections4", "org.apache.commons.collections"] +
+        [
+          ".bidimap;AbstractBidiMapDecorator;true;AbstractBidiMapDecorator;;;MapKey of Argument[0];MapKey of Argument[-1];value",
+          ".bidimap;AbstractBidiMapDecorator;true;AbstractBidiMapDecorator;;;MapValue of Argument[0];MapValue of Argument[-1];value",
+          ".bidimap;AbstractDualBidiMap$BidiMapIterator;true;BidiMapIterator;;;MapKey of Argument[0];MapKey of Argument[-1];value",
+          ".bidimap;AbstractDualBidiMap$BidiMapIterator;true;BidiMapIterator;;;MapValue of Argument[0];MapValue of Argument[-1];value",
+          ".bidimap;AbstractDualBidiMap$EntrySet;true;EntrySet;;;MapKey of Argument[0];MapKey of Element of Argument[-1];value",
+          ".bidimap;AbstractDualBidiMap$EntrySet;true;EntrySet;;;MapValue of Argument[0];MapValue of Element of Argument[-1];value",
+          ".bidimap;AbstractDualBidiMap$EntrySetIterator;true;EntrySetIterator;;;Element of Argument[0];MapKey of Element of Argument[-1];value",
+          ".bidimap;AbstractDualBidiMap$EntrySetIterator;true;EntrySetIterator;;;MapValue of Element of Argument[0];MapValue of Element of Argument[-1];value",
+          ".bidimap;AbstractDualBidiMap$EntrySetIterator;true;EntrySetIterator;;;MapKey of Argument[1];MapKey of Element of Argument[-1];value",
+          ".bidimap;AbstractDualBidiMap$EntrySetIterator;true;EntrySetIterator;;;MapValue of Argument[1];MapValue of Element of Argument[-1];value",
+          ".bidimap;AbstractDualBidiMap$KeySet;true;KeySet;;;MapKey of Argument[0];Element of Argument[-1];value",
+          ".bidimap;AbstractDualBidiMap$KeySetIterator;true;KeySetIterator;;;Element of Argument[0];Element of Argument[-1];value",
+          ".bidimap;AbstractDualBidiMap$KeySetIterator;true;KeySetIterator;;;MapKey of Argument[1];Element of Argument[-1];value",
+          ".bidimap;AbstractDualBidiMap$MapEntry;true;MapEntry;;;MapKey of Argument[0];MapKey of Argument[-1];value",
+          ".bidimap;AbstractDualBidiMap$MapEntry;true;MapEntry;;;MapValue of Argument[0];MapValue of Argument[-1];value",
+          ".bidimap;AbstractDualBidiMap$Values;true;Values;;;MapValue of Argument[0];Element of Argument[-1];value",
+          ".bidimap;AbstractDualBidiMap$ValuesIterator;true;ValuesIterator;;;Element of Argument[0];Element of Argument[-1];value",
```

Comment:
We should be consistent about propagating or not from the `AbstractDualBidiMap<?,V> parent` parameter (EntrySetIterator currently does but ValuesIterator does not, for example). We probably should propagate (though it doesn't matter a lot since these are protected constructors and will be a pain to test also)